gradle-plugin/src/test/java/org/dmfs/gradle/gver/tasks/TagTaskTest.java

```diff
@@ -1,88 +1,72 @@
-
-            // we use a separate gradle home dir to make this test pass
-            (tempDir, repo) -> withResources("temporary user home dir and project",
-                new TempDir(),
-                (FragileFunction<File, WithResource.Resource<Project>, Exception>) new TestProject(tempDir,
-                    new Strategy(
-                        MAJOR.when(new CommitMessage(new Contains("#major"))),
-                        MINOR.when(new CommitMessage(new Contains("#minor"))),
-                        PATCH.when(new CommitMessage(new Contains("#patch"))),
-                        UNKNOWN.when(((repository1, commit, branches) -> true))),
-                    new Suffixes((repository12, commit, branch) -> new Present<>("-SNAPSHOT"))),
 
-                (userHome, project) -> assertionThat(repository -> ((TagTask) project.getTasks().getByName("gitTag")).perform(),
-                    is(procedureThatAffects(
-                        new Text("alters repository"),
-                        () -> repo,
-                        soIt(has(repository -> new Mapped<>(Ref::getName, new Git(repository).tagList().call()),
-                            iteratesInAnyOrder(R_TAGS + "0.0.1", R_TAGS + "0.0.2-alpha.1-SNAPSHOT"))))))));
+    Assertion gitTagRelease_adds_tag_for_current_version =
+        withResource(testProject,
+            project -> assertionThat(repository -> ((TagTask) project.getTasks().getByName("gitTag")).perform(),
+                is(procedureThatAffects(
+                    new Text("alters repository"),
+                    () -> testRepository.value(),
+                    soIt(new HasTagListThat(iteratesInAnyOrder(R_TAGS + "0.0.1", R_TAGS + "0.0.2-alpha.1-SNAPSHOT")))))));
 
 
     Assertion gitTagRelease_fails_on_dirty_repo =
-        withResource("with temporary directory and git repository",
-            new TempDir(),
-            tempDir -> withResource(initialized(repo -> {
-                        new File(tempDir, "newfile").createNewFile();
-                        new Git(repo).add().addFilepattern("newfile").call();
-                    },
-                    () -> new Repository(getClass().getClassLoader().getResource("0.0.2-alpha.1.bundle"), "main").value(tempDir)),
-
-                // we use a separate gradle home dir to make this test pass
-                repo -> withResources("temporary user home dir and project",
-                    new TempDir(),
-                    (FragileFunction<File, WithResource.Resource<Project>, Exception>) new TestProject(tempDir,
-                        new Strategy(
-                            MAJOR.when(new CommitMessage(new Contains("#major"))),
-                            MINOR.when(new CommitMessage(new Contains("#minor"))),
-                            PATCH.when(new CommitMessage(new Contains("#patch"))),
-                            UNKNOWN.when(((repository1, commit, branches) -> true)))),
+        withResource(initialized(
```

Comment:
@sschuberth this test has now shrunk from [the initial 41 lines](https://github.com/dmfs/gver/blob/e1dc9baf844b706c27390b4b5b1248f878a84151/gradle-plugin/src/test/java/org/dmfs/gradle/gver/tasks/TagTaskTest.java#L59-L100) to just 12 lines. I think this should be much ore comprehensible.